app/controllers/users_controller.rb

```diff
@@ -0,0 +1,101 @@
+class UsersController < ApplicationController
+  skip_before_action :require_login, only: [:upvote] #g
+
+  def index
+    @users = User.all
+  end
+
+  def show
+    @user = User.find_by(id: params[:id])
+    return :not_found if @user.nil?
```

Comment:
You need to use the `head` method here or else Rails will try to render the view.

```suggestion
    return head :not_found if @user.nil?
```